serde/de.py

```diff
@@ -379,7 +379,7 @@ def opt(self, arg: DeField) -> str:
                 exists = f'{arg.data} is not None'
             else:
                 exists = f'{arg.datavar}.get("{arg.name}") is not None'
-            return f'{self.render(value)} if {exists} else None'
+            return f'({self.render(value)}) if {exists} else None'
```

Comment:
Yes, to ensure the evaluation order of the potential nested if else statements. Otherwise the `None` check does not work correctly when the output of `render()` is for a complex type. For example for `Path`.

Without the parenthesis the generated code would look like this:
````python
(data["t"] if isinstance(data["t"], Path) else Path(data["t"]))
if reuse_instances
else Path(data["t"])
if data.get("t") is not None
else None
````
When `reuse_instances` is `True` the None check is skipped and the first line is executed directly, resulting in a TypeError when `data["t"]` equals `None`.

With the parenthesis the code looks like this:
````python
(
    (data["t"] if isinstance(data["t"], Path) else Path(data["t"]))
    if reuse_instances
    else Path(data["t"])
)
if data.get("t") is not None
else None
````
Here the `None` check is executed first and would correctly return `None` if `data["t"]` equals `None`.

Reply:
👍